discord/asset.py

```diff
@@ -124,6 +125,39 @@ async def save(self, fp: Union[str, bytes, os.PathLike[Any], io.BufferedIOBase],
             with open(fp, 'wb') as f:
                 return f.write(data)
 
+    async def to_file(self, *, spoiler: bool = False) -> File:
+        """|coro|
+
+        Converts the asset into a :class:`File` suitable for sending via
+        :meth:`abc.Messageable.send`.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        -----------
+        spoiler: :class:`bool`
+            Whether the file is a spoiler.
+
+        Raises
+        ------
+        HTTPException
+            Downloading the asset failed.
```

Comment:
Sorry, I forgot one more thing.
```suggestion
        DiscordException
            The asset does not have an associated state.
        HTTPException
            Downloading the asset failed.
```

Reply:
Also maybe TypeError if it's a sticker of type "lottie"

Reply:
Yup that'd be another case of error.